File: core/src/main/java/com/andersmurphy/geoquiz/presenters/CheatPage.java
package com.andersmurphy.geoquiz.presenters;

public final class CheatPage implements CheatPresenter {

    private final CheatView view;
    private final boolean isQuestionAnswerTrue;
    private boolean hasAnswerBeenShown;

    public CheatPage(CheatView view, boolean isQuestionAnswerTrue, boolean hasAnswerBeenShown) {
        this.view = view;
        this.isQuestionAnswerTrue = isQuestionAnswerTrue;
        this.hasAnswerBeenShown = hasAnswerBeenShown;
    }

    @Override
    public void showAnswerPressed() {
        hasAnswerBeenShown = true;
        if(isQuestionAnswerTrue) {
            view.showAnswerIsTrue();
        } else {
            view.showAnswerIsFalse();
        }
    }

    @Override
    public void setResult() {
        view.setResultAnswerShown(hasAnswerBeenShown);
    }

    @Override
    public void saveState() {
        view.saveAnswerShown(hasAnswerBeenShown);
    }
}

File: core/src/main/java/com/andersmurphy/geoquiz/presenters/CheatPage.kt
package com.andersmurphy.geoquiz.presenters

class CheatPage(private val view: CheatView,
                       private val isQuestionAnswerTrue: Boolean,
                       private var hasAnswerBeenShown: Boolean) : CheatPresenter {

    override fun showAnswerPressed() {
        hasAnswerBeenShown = true
        if (isQuestionAnswerTrue) view.showAnswerIsTrue() else view.showAnswerIsFalse()
    }

    override fun setResult() {
        view.setResultAnswerShown(hasAnswerBeenShown)
    }

    override fun saveState() {
        view.saveAnswerShown(hasAnswerBeenShown)
    }
}
